patches/third_party-blink-renderer-bindings-core-v8-v8_script_runner.cc.patch

```diff
@@ -0,0 +1,22 @@
+diff --git a/third_party/blink/renderer/bindings/core/v8/v8_script_runner.cc b/third_party/blink/renderer/bindings/core/v8/v8_script_runner.cc
+index 919c1b6e3d5c9f5388f0396d5e4f916e3c9e4c6f..9881b544de41d41462addc88361ce3e69f7e3939 100644
+--- a/third_party/blink/renderer/bindings/core/v8/v8_script_runner.cc
++++ b/third_party/blink/renderer/bindings/core/v8/v8_script_runner.cc
+@@ -223,7 +223,7 @@ int GetMicrotasksScopeDepth(v8::Isolate* isolate,
+ 
+ }  // namespace
+ 
+-v8::MaybeLocal<v8::Script> V8ScriptRunner::CompileScript(
```

Comment:
can we do something with `ScriptCompiler::Compile` instead?

Reply:
I don't think so.
1. `ScriptCompiler::Compile*` are v8-level functions.
2. To properly handle all cases I need to patch all `ScriptCompiler::Compile*` variations (including `CompileModule`, etc.).

Reply:
sorry, it would actually be `GetScriptDetails` which gets called in `CompileStreamedSource` and `CompileUnboundInternal`, but it also gets called in some other places so maybe it does work for you. That being said, I'd prefer a couple of `BRAVE_` patches in ScriptSource to making these new static methods

Reply:
`GetScriptDetails` cannot be used, because it's an intermediate step when a script is not compiled yet. I need the data that was used to compile the script and the resulting script to get its id.

> That being said, I'd prefer a couple of BRAVE_ patches in ScriptSource to making these new static methods

I tried `BRAVE_` patch approach to split methods in two (I need to call upstream version and then use the result) , but defines are pretty heavy and ugly. Clang-formatted file: https://gist.github.com/goodov/321f7f258408695c02bead885f357704

Is this what you had in mind or can we do something else?